src/components/QrDataProvider.js

```diff
@@ -81,6 +93,11 @@ const reducer = (state, action) => {
                 ? results.some((series) => series.validPrimarySeries) : null,
               error: null
             };
+            const matchingDemographicData = demographicData.every(
+              (card) => JSON.stringify(card) === JSON.stringify(demographicData[0])
```

Comment:
This is not going to work consistently - the order of fields in the name object might be different between cards. Stringify will only work if they are always in the same order, which is not guaranteed by the FHIR spec. So name: {given: 'A', family: ['B']} will not match name: {family:['B'],given:'A'}, even though they are the same. You could manually push each data element allowed in the spec separately, see https://build.fhir.org/ig/HL7/fhir-shc-vaccination-ig/StructureDefinition-shc-patient-general-dm.html.

Reply:
Yes would check text as well. The IG has constraints that only text or family/given are provided, not both. There are some countries where they use text because names don't fit neatly in family/given (I believe Japan is using text only). If one card has text and another family/given, we should still flag it as a possible mismatch so the person has to manually review.

Reply:
Would also use all the given names, not just first, since we do display them in the UI. You can use patientResource.resource.name[0].given.join(' ') to do a simple string compare. The given names are expected to be serialized in consistent order (first, then middle, etc, see http://hl7.org/fhir/R4/datatypes.html#HumanName).